src/gsi_ncdiag/gsi_ncdiag.py

```diff
@@ -926,9 +926,9 @@ def toIODAobs(self, OutDir, TotalBias=False, clobber=True, platforms=None):
 
                 if v == 'bend':
                     # sort record_number
-                    record_number = self.var('record_number')[idx]
+                    record_number = self.var('Observation_Subtype')[idx]
```

Comment:
and @XuanliLi-NOAA here it is!!!
https://github.com/JCSDA-internal/ioda-converters/blob/f02f792dc42ad78dbecf9f83ed8fd6ee3ffffb8e/src/gsi_ncdiag/gsi_ncdiag.py#L119

yes indeed this is what the sort should be on.. I do not know if or what changes are needed in the `geovals` section as do not have any of those files for testing